OpenRA.Mods.Common/Traits/Repairable.cs

```diff
@@ -93,20 +92,56 @@ public void ResolveOrder(Actor self, Order order)
 
 				self.CancelActivity();
 				self.QueueActivity(new MoveAdjacentTo(self, target));
-				self.QueueActivity(movement.MoveTo(self.World.Map.CellContaining(order.TargetActor.CenterPosition), order.TargetActor));
-				if (CanRearmAt(order.TargetActor) && CanRearm())
-					self.QueueActivity(new Rearm(self));
-
-				self.QueueActivity(new Repair(order.TargetActor));
-
-				var rp = order.TargetActor.TraitOrDefault<RallyPoint>();
-				if (rp != null)
-					self.QueueActivity(new CallFunc(() =>
-					{
-						self.SetTargetLine(Target.FromCell(self.World, rp.Location), Color.Green);
-						self.QueueActivity(movement.MoveTo(rp.Location, order.TargetActor));
-					}));
+				self.QueueActivity(new CallFunc(() => AfterReachActivities(self, order, movement)));
+
+				TryCallTransport(self, target, new CallFunc(() => AfterReachActivities(self, order, movement)));
```

Comment:
This doesn't get cancelled if you tell the unit to do something after the repair.  The carryall will ignore whatever it is now doing and carry it off.